### ISSUE TITLE
clang-format-16 `AlignTrailingComments: Kind: Leave` moves comments #2

### ISSUE THREAD

**Issue by MartinsSmirnovs** [llvm/llvm-project#59635]
Given following `.clang-format` file:
```yaml
AlignTrailingComments:
  Kind: Leave
```
And following `foo.cpp` source file:
```cpp
// Comment with trailing spaces in the end       
// Next comment
```
I expect that `clang-format-16 foo.cpp` will not perform formatting, however, it does:
```console
$ clang-format-16 foo.cpp
// Comment with trailing spaces in the end
       // Next comment
```

`clang-format-16` version:
```console
$ clang-format-16 --version
Ubuntu clang-format version 16.0.0 (++20221221042320+05f343fa8afe-1~exp1~20221221042422.633)
```

**Comment by MartinsSmirnovs** [comment]
@mairacanal maybe you already know how to fix that? You have made fix to very similar issue #59203.

**Comment by lanzhiquan** [comment]
Who knows how to fix this BUG? It still occurs at version 16.0.1.

**Comment by MartinsSmirnovs** [comment]
The issue is still present in `Ubuntu clang-format version 16.0.6 (++20230610113458+7cbf1a259152-1~exp1~20230610233511.102)`

**Comment by MartinsSmirnovs** [comment]
The issue is still present in `Ubuntu clang-format version 17.0.0 (++20230827093036+e91ad6b97fcb-1~exp1~20230827093150.32)`

**Comment by owenca** [comment]
It seems that it's been fixed since version 17.0.1.

**Comment by MartinsSmirnovs** [comment]
@owenca the issue can still be reproduced with `Ubuntu clang-format version 17.0.2 (++20230928073122+1b55dc9d94c3-1~exp1~20230928073220.47)`. Perhaps the fix is not yet available in qualification branch from which I install the executable?

**Comment by owenca** [comment]
@MartinsSmirnovs actually, this was a duplicate of #67116, which was fixed in d3f8c88abe2e.

**Comment by MartinsSmirnovs** [comment]
@owenca tested with `Ubuntu clang-format version 18.0.0 (++20230929042255+ab472cd4ea12-1~exp1~20230929042400.1214)`, the issue is no more. Thanks!
